crates/core/src/db/relational_db.rs

```diff
@@ -191,10 +193,19 @@ impl RelationalDB {
     }
 
     #[tracing::instrument(skip_all)]
-    pub fn schema_for_table<'tx>(&self, tx: &'tx MutTx, table_id: TableId) -> Result<Cow<'tx, TableSchema>, DBError> {
+    pub fn schema_for_table_mut<'tx>(
+        &self,
+        tx: &'tx MutTx,
+        table_id: TableId,
+    ) -> Result<Cow<'tx, TableSchema>, DBError> {
         self.inner.schema_for_table_mut_tx(tx, table_id)
     }
 
+    #[tracing::instrument(skip_all)]
+    pub fn schema_for_table<'tx>(&self, tx: &'tx Tx, table_id: TableId) -> Result<Cow<'tx, TableSchema>, DBError> {
```

Comment:
I suppose we could have this take a `TxMode` or something of a similar concept and not duplicate all the functions, but I dunno how I feel about that. I think what you have now is probably better, but thought I'd mention it.

Reply:
I thought about that as well but then all callers have to keep wrapping tx objects in`TxMode` to pass as param, Right now, `TxMode` is local to `vm.rs` only.
I felt bit of wrapper methods duplicacy is fine here to avoid using extra enum everywhere.